pyat/at/tracking/deprecated.py

```diff
@@ -0,0 +1,285 @@
+          Default for linux is ``'fork'``, default for macOS and  Windows is
+          ``'spawn'``. ``'fork'`` may be used on macOS to speed up the
+          calculation or to solve Runtime Errors, however it is considered
+          unsafe.
+
+    Keyword arguments:
+        keep_lattice (bool):    Use elements persisted from a previous
+          call. If :py:obj:`True`, assume that the lattice has not changed
+          since the previous call.
+        keep_counter (bool):    Keep the turn number from the previous
+          call.
+        turn (int):             Starting turn number. Ignored if
+          *keep_counter* is :py:obj:`True`. The turn number is necessary to
+          compute the absolute path length used in RFCavityPass.
+        losses (bool):          Boolean to activate loss maps output
+        omp_num_threads (int):  Number of OpenMP threads
+          (default: automatic)
+
+    The following keyword arguments overload the Lattice values
+
+    Keyword arguments:
+        particle (Particle):    circulating particle.
+          Default: *lattice.particle* if existing,
+          otherwise *Particle('relativistic')*
+        energy (float):         lattice energy. Default 0.
+
+    If *energy* is not available, relativistic tracking if forced,
+    *rest_energy* is ignored.
+
+    Returns:
+        r_out: (6, N, R, T) array containing output coordinates of N particles
+          at R reference points for T turns.
+        loss_map: If *losses* is :py:obj:`True`: dictionary with the
+          following key:
+
+          ==============    ===================================================
+          **islost**        (npart,) bool array indicating lost particles
+          **turn**          (npart,) int array indicating the turn at
+                            which the particle is lost
+          **element**       ((npart,) int array indicating the element at
+                            which the particle is lost
+          **coord**         (6, npart) float array giving the coordinates at
+                            which the particle is lost (zero for surviving
+                            particles)
+          ==============    ===================================================
+
+    .. note::
+
+       * For multiparticle tracking with large number of turn the size of
+         *r_out* may increase excessively. To avoid memory issues
+         :pycode:`lattice_pass(lattice, r_in, refpts=[])` can be used.
+         An empty list is returned and the tracking results of the last turn
+         are stored in *r_in*.
+       * By default, :py:func:`patpass` will use all the available CPUs.
+         To change the number of cores used in ALL functions using
+         :py:func:`patpass` (:py:mod:`~at.acceptance.acceptance` module for
+         example) it is possible to set ``at.DConstant.patpass_poolsize``
+         to the desired value.
+
+    """
+    kwargs['use_mp'] = True
```

Comment:
`patpass` calls directly `internal_plpass` which does not accept the `use_mp` argument (it's redundant). This line must be removed.

Reply:
@lfarv should the parameter ncpu 'pyat/at/physics/frequency_maps.py' also be removed, ignored or default to None ? 
For example, 'get_acceptance' does not set an explicit value for the max number of cpus, and I think this was done in order to use all cpus available.

Also, 'ncpu' is just an upper limit, I didn't find a direct way to actually confirm the number of cpus that were used by patpass ( would the new pass methods allow to know ?). It seems that pyat prefers to use all cpus and let the user do the settings outside pyat, for example through slurm.

Reply:
Looking at 'pyat/at/physics/frequency_maps.py', I have the following comments:
- what is called `ncpu` here is sent as `pool_size` to `patpass`. It would have been better called `pool_size` to be consistent.
- this parameter only concerns the python multiprocessing library. It's the [number of python processes](https://docs.python.org/3.9/library/multiprocessing.html#module-multiprocessing.pool) started, independently of the real number of processors. It's the exact number of processes, not a maximum. If there are less processors, processes are started sequentially. If `pool_size`/`ncpu` is None, python chooses a number equal to the number of processors. So as for `patpass`, I would set the default to None.
-  This tuning is independent of the users's choice to use slurm,  MPI, OpenMP or other. Though it's not recommend to use several systems together. That's why I would suggest to add a `use_mp` keyword to let the user choose to use python multiprocessing or not.